src/EFRepository.Generator/ExtensionMethodGenerator.cs

```diff
@@ -1,237 +1,388 @@
+		var shortSymbol = compilation.GetTypeByMetadataName("System.Int16");
+		var intSymbol = compilation.GetTypeByMetadataName("System.Int32");
+		var longSymbol = compilation.GetTypeByMetadataName("System.Int64");
+		var floatSymbol = compilation.GetTypeByMetadataName("System.Single");
+		var doubleSymbol = compilation.GetTypeByMetadataName("System.Double");
+		var decimalSymbol = compilation.GetTypeByMetadataName("System.Decimal");
+		var stringSymbol = compilation.GetTypeByMetadataName("System.String");
+		var dateTimeSymbol = compilation.GetTypeByMetadataName("System.DateTime");
+
+		var members = dbSetClass.GetMembers()
+		.Where((m) => !m.IsStatic && !m.IsAbstract && !m.IsImplicitlyDeclared && m.IsDefinition
+					&& m.Kind is SymbolKind.Property or SymbolKind.Field);
+
+		foreach (var member in members)
+		{
+			ITypeSymbol type;
+			bool nullable;
+
+			if (member.Kind is SymbolKind.Property)
+			{
+				var property = (IPropertySymbol)member;
+				type = property.Type;
+				nullable = property.NullableAnnotation == NullableAnnotation.Annotated;
+			}
+			else
+			{
+				var field = (IFieldSymbol)member;
+				type = field.Type;
+				nullable = field.NullableAnnotation == NullableAnnotation.Annotated;
+			}
+
+			if (type.Equals(boolSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateMethod("bool", member.Name, nullable));
+			}
+			else if (type.Equals(byteSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateMethod("byte", member.Name, nullable));
+			}
+			else if (type.Equals(shortSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateMethod("short", member.Name, nullable));
+			}
+			else if (type.Equals(intSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateMethod("int", member.Name, nullable));
+			}
+			else if (type.Equals(longSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateMethod("long", member.Name, nullable));
+			}
+			else if (type.Equals(floatSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateMethod("float", member.Name, nullable));
+			}
+			else if (type.Equals(doubleSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateMethod("double", member.Name, nullable));
+			}
+			else if (type.Equals(decimalSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateMethod("decimal", member.Name, nullable));
+			}
+			else if (type.Equals(stringSymbol, SymbolEqualityComparer.Default))
+			{
+				builder.AppendLine(CreateNullMethodFunctions(member.Name));
+
+				builder.AppendLine($@"
+	/// <summary>
+	/// Filter the <see cref=""IQueryable""/> of {dbSetClass.Name} by {member.Name}
+	/// </summary>
+	/// <param name=""value"">The {type} which {member.Name} should be equal</param>
+	public static IQueryable<{dbSetClass.Name}> By{member.Name}(this IQueryable<{dbSetClass.Name}> query, {type}? value)

```

Comment:
You're using the type variable here, but we know the type to be a string at this point. It probably doesn't matter, but was there a specific reason for this?

Reply:
Could we use that same `nullable` variable here to determine if the string should be nullable? At minimum this can cause problems with the full framework, since strings are always nullable there. And this project still targets full framework which doesn't support the whole `EnableNullable` thing.

Reply:
Now that I've seen the `CreateNullMethodFunctions` method, shouldn't we also use that for `string`? Those can be null as well, and we may want to filter by that.

Reply:
I tried to write the string functions in such a way that they can be nullable or not and we handle them appropriately. If they are null, a null check is performed thus resolving both the new and old styles.

Reply:
The string is a special case in which I want to use the "IsNullOrWhitespace" function instead of simply checking for null

Reply:
It is generating the IsNull and IsNotNull functions, but I think I'd like to add a couple more of it makes sense. By{field}IsNullOrWhiteSpace() and By{field}IsNotNullOrWhiteSpace()

Thoughts on that?